src/smt/proof_manager.cpp

```diff
@@ -105,6 +105,20 @@ PfManager::PfManager(Env& env)
 
 PfManager::~PfManager() {}
 
+// Remove in favor of `std::erase_if` with C++ 20+.
+#if __cplusplus <= 202002L
```

Comment:
I added this pragma to force the compiler to error if/when we move to C++ 20+ (in which case, we should use the standard's version).